crest/Assets/Crest/Crest/Scripts/LodData/RegisterClipSurfaceInput.cs

```diff
@@ -98,9 +235,80 @@ private void LateUpdate()
         protected override bool FeatureEnabled(OceanRenderer ocean) => ocean.CreateClipSurfaceData;
         protected override string RequiredShaderKeywordProperty => LodDataMgrClipSurface.MATERIAL_KEYWORD_PROPERTY;
         protected override string RequiredShaderKeyword => LodDataMgrClipSurface.MATERIAL_KEYWORD;
-
         protected override string MaterialFeatureDisabledError => LodDataMgrClipSurface.ERROR_MATERIAL_KEYWORD_MISSING;
         protected override string MaterialFeatureDisabledFix => LodDataMgrClipSurface.ERROR_MATERIAL_KEYWORD_MISSING_FIX;
+
+        protected override bool RendererRequired => _mode == Mode.Geometry;
+        protected override bool RendererOptional => _mode != Mode.Geometry;
+
+        // Use Unity's UV sphere mesh for gizmos as Gizmos.DrawSphere is too low resolution.
+        static Mesh s_SphereMesh;
+
+        protected new void OnDrawGizmosSelected()
+        {
+            Gizmos.color = GizmoColor;
+
+            if (_mode == Mode.Geometry)
+            {
+                if (TryGetComponent<MeshFilter>(out var mf))
+                {
+                    Gizmos.DrawWireMesh(mf.sharedMesh, 0, transform.position, transform.rotation, transform.lossyScale);
+                }
+
+                return;
+            }
+
+            if (s_Quad == null)
+            {
+                s_Quad = Resources.GetBuiltinResource<Mesh>("Quad.fbx");
+            }
+
+            // Show gizmo for quad which encompasses the shape.
+            Gizmos.matrix = QuadMatrix;
+            Gizmos.DrawWireMesh(s_Quad);
+
+            Gizmos.matrix = transform.localToWorldMatrix;
+
+            switch (_primitive)
+            {
+                case Primitive.Sphere:
+                    if (s_SphereMesh == null)
+                    {
+                        s_SphereMesh = Resources.GetBuiltinResource<Mesh>("New-Sphere.fbx");
+                    }
+
+                    // Render mesh and wire sphere at default size (0.5m radius) which is scaled by gizmo matrix.
+                    Gizmos.DrawMesh(s_SphereMesh, submeshIndex: 0, Vector3.zero, Quaternion.identity, Vector3.one);
+                    Gizmos.DrawWireSphere(Vector3.zero, 0.5f);
+                    break;
+                case Primitive.Cube:
+                    // Render mesh and wire box at default size which is scaled by gizmo matrix.
+                    Gizmos.DrawCube(Vector3.zero, Vector3.one);
+                    Gizmos.DrawWireCube(Vector3.zero, Vector3.one);
+                    break;
+                default:
+                    Debug.LogError("Crest: Not a valid primitive type!");
+                    break;
+            }
+        }
 #endif
     }
+
+    public partial class RegisterClipSurfaceInput : ISerializationCallbackReceiver
```

Comment:
oh it was probably already your plan but may be worth adding a comment above this just saying 

`// Version handling - perform data migration after data loaded` 

or whatever makes sense here

Reply:
Done.